Libraries/Lists/VirtualizedList.js

```diff
@@ -1371,30 +1369,48 @@ class VirtualizedList extends React.PureComponent<Props, State> {
     return !this.props.horizontal ? metrics.y : metrics.x;
   }
 
-  _maybeCallOnEndReached() {
-    const {
-      data,
-      getItemCount,
-      onEndReached,
-      onEndReachedThreshold,
-    } = this.props;
-    const {contentLength, visibleLength, offset} = this._scrollMetrics;
-    const distanceFromEnd = contentLength - visibleLength - offset;
+  _maybeCallOnEndReached(hasShrinkedContentLength: boolean = false) {
+    const {onEndReached, onEndReachedThreshold} = this.props;
+    if (!onEndReached) {
+      return;
+    }
+
+    const {contentLength, visibleLength, offset, dOffset} = this._scrollMetrics;
+
+    // If this is just after the initial rendering
     if (
-      onEndReached &&
-      this.state.last === getItemCount(data) - 1 &&
-      /* $FlowFixMe(>=0.63.0 site=react_native_fb) This comment suppresses an
-       * error found when Flow v0.63 was deployed. To see the error delete this
-       * comment and run Flow. */
-      distanceFromEnd < onEndReachedThreshold * visibleLength &&
-      (this._hasDataChangedSinceEndReached ||
-        this._scrollMetrics.contentLength !== this._sentEndForContentLength)
+      !hasShrinkedContentLength &&
+      !this._hasDoneFirstScroll &&
+      offset === 0
     ) {
-      // Only call onEndReached once for a given dataset + content length.
-      this._hasDataChangedSinceEndReached = false;
-      this._sentEndForContentLength = this._scrollMetrics.contentLength;
-      onEndReached({distanceFromEnd});
+      return;
     }
+
+    // If scrolled up in the vertical list
+    if (dOffset < 0) {
+      return;
+    }
+
+    // If contentLength has not changed
+    if (contentLength === this._sentEndForContentLength) {
+      return;
+    }
+
+    const distanceFromEnd = contentLength - visibleLength - offset;
+
+    // If the distance is so farther than the area shown on the screen
+    if (distanceFromEnd >= visibleLength * 1.5) {
```

Comment:
magic number 1.5

Reply:
`_maybeCallOnEndReached` will just `return` if `distanceFromEnd` is `1.5` times higher then then `visibleLength`. I believe currently it is represented with the [`onEndReachedThreeshold`](https://reactnative.dev/docs/flatlist#onendreachedthreshold).

He includes same behaviour in the lines below.

```
const minimumDistanceFromEnd = onEndReachedThreshold * visibleLength;
if (distanceFromEnd >= minimumDistanceFromEnd) {
   return;
}
```

I believe his implementation accepts minimum `onEndReachedThreeshold` to be 1.5 times the `visibleLenght`, as for any value under `1.5 * visibleLength`, the above statement will be triggered